tap_partnerize/client.py

```diff
@@ -134,18 +183,36 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
         Yields:
             Each record from the source.
         """
-        # TODO: Parse response body and return a set of records.
-        yield from extract_jsonpath(self.records_jsonpath, input=response.json())
 
-    def post_process(self, row: dict, context: dict | None = None) -> dict | None:
-        """As needed, append or transform raw data to match expected structure.
-
-        Args:
-            row: An individual record from the stream.
-            context: The stream context.
-
-        Returns:
-            The updated record dictionary, or ``None`` to skip the record.
-        """
-        # TODO: Delete this method if not needed.
-        return row
+        count = 0
+        f = io.StringIO(response.text)
+        for row in csv.DictReader(f):
+            count += 1
+            if row.get("meta_conversion_gross_value") == "undefined":
+                row.pop("meta_conversion_gross_value")
+            row["meta_conversion_gross_value"] = set_none_or_cast(row.get("meta_conversion_gross_value"), float)
+            if row.get("meta_item_product_id") == "undefined":
+                row.pop("meta_item_product_id")
+            row["item_value"] = set_none_or_cast(row.get("item_value"), float)
+            row["conversion_lag"] = set_none_or_cast(row.get("conversion_lag"), int)
+            row["meta_conversion_delivery_cost"] = set_none_or_cast(row.get("meta_conversion_delivery_cost"), float)
+            row["creative_type"] = set_none_or_cast(row.get("creative_type"), int)
+            row["item_publisher_commission"] = set_none_or_cast(row.get("item_publisher_commission"), float)
+            row["publisher_commission"] = set_none_or_cast(row.get("publisher_commission"), float)
+            row["value"] = set_none_or_cast(row.get("value"), float)
+            row["quantity"] = set_none_or_cast(row.get("quantity"), int)
+            row["meta_conversion_container_version"] = set_none_or_cast(row.get("meta_conversion_container_version"), int)
+            if row.get("job_id"):
+                continue
+            try:
+                row.pop(None, None)
```

Comment:
🤔

Reply:
This was also there to maintain parity with the [existing Airflow job](https://github.com/nymag/airflow/blob/master/airflow/dags/import_partnerize.py#L104), so I'm not _quite_ sure, but it looks like this is meant to omit rows that don't align with a header in the CSV, so basically, misformatted rows or rows with more values than columns in the CSV. But maybe @yujoy can help validate that claim too.

Reply:
Yeah I think it could be simplified to `row.pop(None)`, but basically I think you're right. I think it will only find a key of "None" when there are more values in a row than headers. (like a `"": "value"` dictionary entry). But I'll take a fresh look at this as I work on the post_process method.

Reply:
Just ran a test and ran into an error while iterating through the dictionary entries: `AttributeError: 'NoneType' object has no attribute 'replace'`
So the `row.pop(None, None)` is just to eliminate any values from the dictionary that aren't associated with a key, for whatever reason. They don't seem to be part of our schema either way, so I don't think there's a concern there, but I can add a comment to that affect.